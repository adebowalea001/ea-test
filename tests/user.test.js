import React from 'react';
import sinon from 'sinon';
import { shallow, mount } from 'enzyme';
import User from '../react/components/user.jsx';
import axios from 'axios';

function handleClick() {
  return true;
}

describe('User', () => {

  it('-- should render <User /> component', () => {
    const handleClick = sinon.spy();
    const component = shallow(<User handleClick={ handleClick } />);
    expect(component.getElements()).toMatchSnapshot();
  });

  it('-- should validate button click', () => {
    const handleClick = sinon.spy();
    const component = shallow(<User handleClick={ handleClick } />);
    component.find('button').simulate('click');
    expect(handleClick).toHaveProperty('callCount', 1);
  });

  it('-- should validate user data', done => {
    const handleClick = (cb) => {
      cb(null, {
        username: "Amilea",
        contract: "Consultant",
        country: "uk"
      });


      setTimeout(() => {
           expect(component.ref('username').innerHTML).toEqual('Amilea');
           expect(component.ref('consultant').getAttribute('value')).toEqual('true');
           expect(component.ref('country').value).toEqual('uk');
           done();
       }, 0);
    }

    const component = mount(<User handleClick={ handleClick } />);
    component.find('button').simulate('click');
  });

});
