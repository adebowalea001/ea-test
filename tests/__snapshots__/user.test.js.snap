// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User -- should render <User /> component 1`] = `
Array [
  <div
    className="main"
  >
    <button
      className="btn btn-success"
      onClick={[Function]}
    >
      Show User Details
    </button>
    <div
      className="clearfix"
    />
    <div
      className="titles"
    >
      Name:
    </div>
    <div
      className="user_info username"
    >
      <span
        username={undefined}
      />
    </div>
    <div
      className="titles"
    >
      Type of Contract:
    </div>
    <div
      className="user_info"
    >
      Consultant: 
      <input
        checked={false}
        name="contract"
        onChange={[Function]}
        type="radio"
        value="false"
      />
      Â  Parmanent: 
      <input
        checked={false}
        name="contract"
        onChange={[Function]}
        type="radio"
        value="false"
      />
    </div>
    <div
      className="titles"
    >
      Country of Residence:
    </div>
    <div
      className="user_info"
    >
      <select
        value={undefined}
      >
        <option
          value="ae"
        >
          United Arab Emirates
        </option>
        <option
          value="ng"
        >
          Nigeria
        </option>
        <option
          value="uk"
        >
          United Kingdom
        </option>
        <option
          value="us"
        >
          United States of America
        </option>
      </select>
    </div>
  </div>,
]
`;
